In [1]:
import spacy
import re
from collections import Counter

In [2]:
nlp = spacy.load('en_core_web_sm')
nlp.disable_pipes('tagger', 'parser', 'lemmatizer', 'ner')

['tagger', 'parser', 'lemmatizer', 'ner']

In [3]:
# Task 1

f1 = open('DS3_C2_S2_48hrs_Data_Practice.txt','r').read()
f2 = open('DS3_C2_S2_Scifiscripts_Data_Practice.txt', 'r').read()
f3 = open('DS3_C2_S2_TenThingsIHate_Data_Practice.txt','r').read()

In [4]:
f1 = nlp(f1)
f2 = nlp(f2)
f3 = nlp(f3)

In [5]:
count48 = Counter()

In [6]:
for token in f1:
    count48[token.orth_]+=1

In [7]:
type(count48)

collections.Counter

In [8]:
count48

Counter({'\n\t\t\t\t\t': 1,
         '"': 10,
         '48': 3,
         'HRS': 1,
         '.': 2050,
         '\n\n\t\t\t\t\t   ': 2,
         'by': 45,
         '\n\n\t\t\t\t   ': 2,
         'Roger': 1,
         'Spottiswoode': 1,
         'and': 274,
         '\n\t\t\t\t   ': 2,
         'Walter': 1,
         'Hill': 2,
         '&': 2,
         'Larry': 1,
         'Gross': 1,
         'Steven': 1,
         'E.': 1,
         'De': 1,
         'Souza': 1,
         '\n\n': 3,
         'FADE': 1,
         'IN': 2,
         ':': 1,
         'OPEN': 1,
         'COUNTRY': 1,
         '-': 116,
         'DAY': 8,
         '\n\n\t': 903,
         'Endless': 1,
         'green': 4,
         'hills': 1,
         'bisected': 1,
         'a': 545,
         'ribbon': 1,
         'of': 265,
         'highway': 1,
         'A': 60,
         'road': 9,
         '\n\t': 478,
         'gang': 9,
         'clearing': 1,
         'brush': 1,
         'the': 956,
         'side': 26,
         '...':

In [9]:
# task 2

count48.most_common(10)

[('.', 2050),
 ('\n\t\t\t', 2025),
 ('the', 956),
 ('\n\n\t\t\t\t\t', 937),
 (',', 929),
 ('\n\n\t', 903),
 ('...', 549),
 ('a', 545),
 ('\n\t', 478),
 ('I', 458)]

"." appears the most with 2050 occurances

In [10]:
samp48hrs = f1[:1500]

In [11]:
type(count48)

collections.Counter

In [12]:
prisonercount = dict(count48).get('prisoner') + dict(count48).get('Prisoner')
prisonercount

5

In [13]:
prisonerscount = dict(count48).get('prisoners') + dict(count48).get('Prisoners')
prisonerscount

2

Prisoner appeared a total of 5 times and Prisoners appeared a total of 2 times including lowercase and uppercase

In [14]:
# task 3

temp = nlp('John went to the U.S.A to visit his elder sister.')

In [15]:
counttemp = Counter()
for token in temp:
    counttemp[token.orth_]+=1

In [16]:
counttemp, sum(counttemp.values())

(Counter({'John': 1,
          'went': 1,
          'to': 2,
          'the': 1,
          'U.S.A': 1,
          'visit': 1,
          'his': 1,
          'elder': 1,
          'sister': 1,
          '.': 1}),
 11)

A total of 11 tokens were produced in the given text and "U.S.A" was considered one single token.

In [17]:
# task 4

countthings = Counter()
countscifi = Counter()

for token in f2:
    countthings[token.orth_]+=1
    
for token in f3:
    countscifi[token.orth_]+=1

In [18]:
thecount48 = dict(count48).get('the') + dict(count48).get('The')
thecountscifi = dict(countscifi).get('the') + dict(countscifi).get('The')
thecountthings = dict(countthings).get('the') + dict(countthings).get('The')

In [19]:
print(f"""Total number of 'The' and 'the' in 48hrs.txt are = {thecount48}
Total number of 'The' and 'the' in scifiscripts.txt are = {thecountscifi}
Total number of 'The' and 'the' in tenthingsihate.txt are = {thecountthings}""")

Total number of 'The' and 'the' in 48hrs.txt are = 1059
Total number of 'The' and 'the' in scifiscripts.txt are = 670
Total number of 'The' and 'the' in tenthingsihate.txt are = 664


As we can see, 'Ten Things I hate' have the least number of The's and the's at just 664